### DISCUSSION TITLE
Sonarqube bugs and code smells

### DISCUSSION THREAD

**Discussion by Serrof** [Hipparchus-Math/hipparchus#320] [General]
The new sonarqube config is giving us a bad mark. Loads of major bugs and code smells.
But I think they are mostly on the test folders, so not critical for the library itself. Could we configure Sonarqube to flag things differently?
@sdinot 

Cheers
Romain.

**Comment by sdinot** [comment]
The problem is probably due to the migration of SonarQube from version 7.5 to 9.9 LTS. I'll investigate tomorrow.

**Comment by sdinot** [comment]
I've just modified the project configuration in SonarQube.

**Comment by sdinot** [comment]
Thanks to @MaximeJo, the issue is resolved. :)

**Reply by MaximeJo** [reply]
Hi,
@Serrof you were right the bugs came from the test suite.

There are still more than 1000 code smells with the new Sonar.
30 of them were tagged as false positives in the previous Sonar, I guess we should resolve them in the same fashion.
There was a typo in the config of "SuppressionCommentFilters", I fixed it.
Also the token in the rule "Unncesseray Parentheses" was not appropriate, I changed that.
As for the remaining code smells I think they should be addressed individually.

Edit: Down to 392 code smells. I fixed some rules that weren't tailored for Hipparchus

**Comment by sdinot** [comment]
Dear all,

After SonarQube was upgraded, the quality of the Hipparchus project seemed to have deteriorated. This was mainly due to changes in SonarQube rules, and SonarQube and Sonar Scanner operation. Maxime and I made a few adjustments to SonarQube and Sonar Scanner configuration to improve the relevance of the qualimetric report.

Only 28 code smells remain to be corrected.

**Reply by Serrof** [reply]
Thanks guys for fixing this
